src/visualization.jl

```diff
@@ -2,78 +2,117 @@ using Catlab.CategoricalAlgebra
 using Catlab.Graphics.Graphviz
 import Catlab.Graphics.Graphviz: Graph, Edge, Subgraph
 import Base.Iterators: flatten
+import Base.:(==)
+import Base.hash
 using StatsBase
 
 export Graph
 
-countmap_wrap(a) = isempty(a) ? Dict{Int, Int}() : countmap(a)
-
-function edgify(δ::Dict{Int64, Int64}, transition, reverse::Bool; pre="")
-  return [Edge(reverse ? ["\"$(pre)t$transition\"", "\"$(pre)s$k\""] :
-                         ["\"$(pre)s$k\"", "\"$(pre)t$transition\""],
-               Attributes(:label=>"$(δ[k])", :labelfontsize=>"6"))
-           for k in collect(keys(δ)) if δ[k] != 0]
-end
-
-function edgify(δ::Dict{Tuple{Int64, Bool}, Int64}, transition, reverse::Bool; pre="", lw=3.0)
-  return [Edge(reverse ? ["\"$(pre)t$transition\"", "\"$(pre)s$(k[1])\""] :
-                         ["\"$(pre)s$(k[1])\"", "\"$(pre)t$transition\""],
-               Attributes(:label=>"$(δ[k])", :labelfontsize=>"6",
-                          :penwidth=>(k[2] ? "$lw" : "1.0")))
-           for k in collect(keys(δ)) if δ[k] != 0]
-end
+==(a::NodeID, b::NodeID) = isequal(a.name, b.name) && isequal(a.port, b.port) && isequal(a.anchor, b.anchor) && true
```

Comment:
@epatters I currently am defining the equality and hash operators for `NodeID` and `Edge` objects here (to allow `countmap` to operate on arrays of `Edge`s. Is this something I should move upstream to `Catlab.Graphics.Graphviz`, or is it alright to just define them here?

Reply:
I think we can leave it here and if it comes up again, copy it into Catlab.

Reply:
I meant to say this earlier but overriding `Base.==` for a type in Catlab is type piracy, so it shouldn't be here. Let's fix this properly by adding the appropriate methods to Catlab.